In [2]:
import pandas as pd

In [31]:
df = pd.read_csv("./vine_table.csv")

In [32]:
df

review_id  star_rating  helpful_votes  total_votes vine  \
0        RMDCHWD0Y5OZ9          3.0            0.0          1.0    N   
1        RZSL0BALIYUNU          5.0            0.0          0.0    N   
2        RIZR67JKUDBI0          3.0            0.0          1.0    N   
3       R27HL570VNL85F          5.0            0.0          0.0    N   
4       R34EBU9QDWJ1GD          5.0            0.0          0.0    N   
...                ...          ...            ...          ...  ...   
904760   R6MDTUK723YPY          4.0          143.0        148.0    N   
904761  R10B4417NCK98O          5.0           24.0         24.0    N   
904762  R3V2A0GUZ3JAQ4          4.0           27.0         29.0    N   
904763  R1VYXC7FFG4LGH          4.0           46.0         49.0    N   
904764  R2XJJXFZOM36V5          5.0           17.0         19.0    N   

       verified_purchase  
0                      N  
1                      Y  
2                      Y  
3                      Y  
4                      Y  
...                  ...  
904760                 N  
904761                 N  
904762                 N  
904763                 N  
904764                 N  

[904765 rows x 6 columns]

In [33]:
filtered_df = df.loc[df['total_votes']>20]
filtered_df.sample(5)

review_id  star_rating  helpful_votes  total_votes vine  \
856852  R1M6MSBXCCETEF          5.0           19.0         24.0    N   
883057  R2C4FY1FSWI0N8          5.0           70.0         72.0    N   
900027  R24TFLO8WIVSYN          3.0           59.0         66.0    N   
879022  R3BT6QUK2ZHL10          5.0           54.0         60.0    N   
570927  R19S0BKXTLX3T3          5.0           53.0         56.0    N   

       verified_purchase  
856852                 Y  
883057                 Y  
900027                 Y  
879022                 Y  
570927                 N

In [34]:
helpful_votes_df = filtered_df.loc[(filtered_df.helpful_votes / filtered_df.total_votes) >= .5]

helpful_votes_df.sample(5)

review_id  star_rating  helpful_votes  total_votes vine  \
880169  R23G10SRQ7LRBM          5.0           46.0         49.0    N   
886105  R21U7TFFMTRESK          2.0           27.0         29.0    N   
514798  R37ZRPIYCDKMHI          5.0           41.0         43.0    N   
873164   R3NCGLR52TA3M          5.0           41.0         43.0    N   
501115  R27OFCIH0VITGE          5.0           38.0         40.0    N   

       verified_purchase  
880169                 Y  
886105                 Y  
514798                 Y  
873164                 Y  
501115                 Y

In [53]:
vine_Y_df = helpful_votes_df.loc[df["vine"]=="Y"]
vine_Y_df.sample(5)


review_id            59
star_rating          59
helpful_votes        59
total_votes          59
vine                 59
verified_purchase    59
dtype: int64

In [48]:
vine_N_df = helpful_votes_df.loc[df["vine"]=="N"]
vine_N_df.sample(5)

review_id  star_rating  helpful_votes  total_votes vine  \
865506  R1C1C0MFZBOU2Z          5.0           20.0         22.0    N   
862067   RHBWH3AL0HE39          1.0           23.0         25.0    N   
835164  R1ST5TJLH3E8VF          5.0           75.0         86.0    N   
527043   RDGHELQX4L9JI          4.0           56.0         58.0    N   
864255  R36BQPE0IZXUGM          5.0           21.0         21.0    N   

       verified_purchase  
865506                 Y  
862067                 N  
835164                 Y  
527043                 N  
864255                 Y

## Determine the number of 5-Star reviews in the Paid reviewers vs unpaid reviewers

### Paid
---

In [64]:
num_Y_reviews = vine_Y_df.shape[0]
num_Y_reviews

59

In [61]:
Y_counts = (vine_Y_df.star_rating==5.0).value_counts()
Y_counts 

True     34
False    25
Name: star_rating, dtype: int64

In [66]:
(Y_counts[0]/num_Y_reviews)*100

42.3728813559322

There are **34 5-Star ratings** in the Paid Vine Reviews. 
**42.37%** of the reivews are 5-Star

###### num_N_reviews = vine_N_df.shape[0]
num_N_reviews

In [70]:
N_counts = (vine_N_df.star_rating==5.0).value_counts()
N_counts 

True     7678
False    5802
Name: star_rating, dtype: int64

In [71]:
(N_counts[0]/num_N_reviews)*100

43.04154302670623

There are **7678** 5-Star ratings in the **unpaid** Vine Reviews. 
**43.04%** of the reivews are 5-Star